HV-991 Add custom weigths parameter to Mod11Check (and Mod10Check)

marko-bekhta opened the pull request:
- https://hibernate.atlassian.net/browse/HV-991

added optional weights parameter to Mod11Check annotation 
added custom annotation for 3 Polish identification numbers - NIP, PESEL, REGON
added tests for new annotations
added checks to annotation processor

gunnarmorling commented on engine/src/main/java/org/hibernate/validator/constraints/pl/REGON.java:
```diff
@@ -0,0 +1,56 @@
+
+
+/**
+ * Validates a REGON number - (9/14 digit number from Rejestr Gospodarki Narodowej ).
```
Could you add an (English) description of "Rejestr Gospodarki Narodowej"?

gunnarmorling commented on engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/pl/NIPValidator.java:
```diff
@@ -0,0 +1,37 @@
+import org.hibernate.validator.constraints.pl.NIP;
+
+/**
+ * A class for NIP validator.
```
"Validator for @NIP", it's apparent that it is a class :)

gunnarmorling commented on engine/src/main/java/org/hibernate/validator/internal/util/ModUtil.java:
```diff
@@ -101,4 +101,33 @@ public static int calculateMod11Check(final List<Integer> digits, final int thre
+	 * @param moduloParam modulo parameter to be used
+	 * @param weights weights for the sum.
+	 *
+	 * @return
```
Better to omit it.

gunnarmorling commented on engine/src/main/java/org/hibernate/validator/constraints/Mod11Check.java:
```diff
@@ -122,6 +122,15 @@
+	 * If the size of provided weights array is less than needed ({@code array.length < endIndex - startIndex})
+	 * the weights will be used repeatedly. For example if constant weight is needed the weights array
+	 * can be initialized as {@code weights = { someConstant }}.
+	 * If weights array is left blank (which is a default value) a classical Mod11 weights are used.
```
How about that: "By default a weight of 1 will be used for each digit".

gunnarmorling replied:
Thinking more about it, is this actually used for the new constraints? If not, I don't think we should add it.

gunnarmorling replied:
If these new constraints cannot be written based on top of the existing ones, I wouldn't bother to add these weights then (and let's change the title of the issue to reflect that it is about adding these three Polish constraints).

gunnarmorling commented on engine/src/main/resources/org/hibernate/validator/ValidationMessages.properties:
```diff
@@ -32,3 +31,8 @@ org.hibernate.validator.constraints.URL.message                     = must be a
 org.hibernate.validator.constraints.br.CPF.message                  = invalid Brazilian individual taxpayer registry number (CPF)
 org.hibernate.validator.constraints.br.TituloEleitoral.message      = invalid Brazilian Voter ID card number
+
+org.hibernate.validator.constraints.pl.REGON.message = Invalid Polish Taxpayer Identification Number (REGON)
```
Seems we don't have a Polish translations file yet. Maybe - as a separate issue -you could create one (assuming you speak the language based on the fact that you created these constraints :) )?

marko-bekhta commented:
I've removed that custom weights parameter and cleaned up JavaDocs. Also I've updated the JIRA ticket. As for the translations - I'll need to create separate JIRA tickets for those, right ?

gunnarmorling commented:
Yes, separate tickets for those changes would be great.

gunnarmorling commented:
Rebased and applied. Thanks, @marko-bekhta! Could you send in another PR for adding the new constraints to the docs as well (see [here](https://github.com/hibernate/hibernate-validator/blob/master/documentation/src/main/asciidoc/ch02.asciidoc#additional-constraints))? Thanks again!

marko-bekhta commented:
@gunnarmorling https://github.com/hibernate/hibernate-validator/pull/567 here's a PR for doc. Is that what you had in mind ? Or should there be more information ? :)
